src/psyclone/tests/psyir/symbols/datatype_test.py

```diff
@@ -835,7 +850,7 @@ def test_unsupported_fortran_type_is_allocatable(fortran_reader):
     vsym = routine.symbol_table.lookup("var")
     # Make sure we do indeed test the UnsupportedFortranType
     assert isinstance(vsym.datatype, UnsupportedFortranType)
-    assert not vsym.datatype.is_allocatable
+    assert vsym.datatype.is_allocatable is False
```

Comment:
Please add a check that this returns `None` if `partial_datatype` is None.

Reply:
Done.